src/NosnocSqpMethod.m

```diff
@@ -0,0 +1,105 @@
+% BSD 2-Clause License
+
+% Copyright (c) 2022, Armin Nurkanović, Jonathan Frey, Anton Pozharskiy, Moritz Diehl
+
+% Redistribution and use in source and binary forms, with or without
+% modification, are permitted provided that the following conditions are met:
+
+% 1. Redistributions of source code must retain the above copyright notice, this
+%    list of conditions and the following disclaimer.
+
+% 2. Redistributions in binary form must reproduce the above copyright notice,
+%    this list of conditions and the following disclaimer in the documentation
+%    and/or other materials provided with the distribution.
+
+% THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+% AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+% IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+% DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+% FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+% DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+% SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+% CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+% OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+% OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+% This file is part of NOSNOC.
+
+classdef NosnocSqpMethod < handle % TODO maybe handle not necessary, revisit.
+    properties
+
+    end
+
+    methods
+        function solver = construct_solver(obj, nlp, solver_options, time_remaining)
+            import casadi.*
+            w = nlp.w;
+            g = nlp.g;
+            p = nlp.p;
+
+            casadi_nlp = struct('f', nlp.augmented_objective, 'x', w, 'g', g, 'p', p);
+
+            opts_casadi_nlp = solver_options.opts_casadi_nlp;
+            opts_casadi_nlp = rmfield(opts_casadi_nlp, 'ipopt');
+            opts_casadi_nlp = rmfield(opts_casadi_nlp, 'snopt');
+            opts_casadi_nlp = rmfield(opts_casadi_nlp, 'worhp');
+            opts_casadi_nlp = rmfield(opts_casadi_nlp, 'uno');
+            opts_casadi_nlp.min_step_size=1e-15;
+            %opts_casadi_nlp.qpsol_options.schur = true;
+
+            warning('sqpmethod is not intended for use and is included only for experimental purposes')
```

Comment:
The plugin class is used to generate a solver object so solve is called on the output of this method.

This warning is here because the `sqpmethod` casadi solver performs very poorly generally.